cmd/agent/app/diagnose.go

```diff
@@ -16,18 +16,40 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var (
+	diagnoseCommand = &cobra.Command{
+		Use:        "diagnose",
+		Short:      "Execute some connectivity diagnosis on your system",
+		Long:       ``,
+		Deprecated: "this command is only here for backward compatibility. Please consider using 'diagnose metadata-availability' instead.",
```

Comment:
Deprecation implies backward compatibility already. Also would be good to mention deprecation policy: is it going to be removed in Agent 7.38? Agent 8.0? Never (in which case we might want to drop deprecation, and simply support it as an alias).

```suggestion
		Deprecated: "Renamed to 'diagnose metadata-availability'.",
```

Reply:
I don't think we will remove this command in the future. Maybe we'll even add behavior to this command to execute both subcommand successively idk 🤷